### ISSUE TITLE
[Poll] should default values also be used when null is found?

### ISSUE THREAD

**Issue by xaviergonz** [xaviergonz/mobx-keystone#44]
Right now default values are only applied when undefined is found, e.g.

`prop(42)` will turn undefined to 42
`tProp(types.number, 42)` will turn undefined to 42

but there's no way to do it for null
`tProp(types.maybeNull(types.number), 42)` will turn undefined to 42, but will leave null as is
`prop<number | null | undefined>(42)` same

so the question is, should a default value transform both undefined AND null into the default value? e.g.

`prop(42)` will turn null and undefined into 42
`tProp(types.number, 42)` will do the same

it would still be ok to have no default value as usual
`tProp(types.maybe(types.number))` both number and undefined are accepted and left as is, null is not valid
`tProp(types.maybeNul(types.number))` both number and null are accepted and left as is, undefined is not valid

**Comment by terrysahaidak** [comment]
It definitely makes sense for me here.
The only case where it might be not so good – you won't ever know your server returned undefined or null in that case. I've been validating my server responses in MST using type-checking, but this is more like an edge case.

**Comment by xaviergonz** [comment]
For tricky edge cases I guess you could just use a computed property. e.g.

// must verify that server returns number or null, use 42 as default value when null
```ts
@model("...")
class M extends Model({
  _x: tProp(types.or(types.number, types.null))
}) {
  @computed
  get x() {
    return this._x === null ? 42 : this._x; // return 42 when null
  }
}
```

// undefined must use default, null must be kept as null
```ts
@model("...")
class M extends Model({
  _x: tProp(types.or(types.number, types.null, types.undefined))
}) {
  @computed
  get x() {
    return this._x === undefined ? 42 : this._x; // return 42 when undefined, null when null
  }
}
```

But I think that's unusual compared to wanting a default value on undefined/null.

**Comment by xaviergonz** [comment]
@sisp thoughts?

**Comment by xaviergonz** [comment]
Also with this change it should be easy to normalize undefined/null into either only undefined or only null

```ts
tProp(types.number, undefined) // number is kept as is, null/undefined becomes undefined
tProp(types.number, null) // number is kept as is, null/undefined becomes null
```

**Comment by terrysahaidak** [comment]
@xaviergonz I'm agree.

**Comment by xaviergonz** [comment]
Will be released as 0.23.0
